src/main/resources/org/broadinstitute/hellbender/utils/plotter/ACNVResultsPlottingPerSeg.R

```diff
@@ -0,0 +1,88 @@
+#NOTE: the Java wrapper for this script first sources CNV_plotting_library.R
+options(error = quote({dump.frames(dumpto = "plotting_dump", to.file = TRUE); q()}))    # Useful for debugging
+
+library(optparse)
+option_list = list(
+    make_option(c("--sample_name", "-sample_name"), dest="sample_name", action="store"),
+    make_option(c("--snp_counts_file", "-snp_counts_file"), dest="snp_counts_file", action="store"),
+    make_option(c("--coverage_file", "-coverage_file"), dest="coverage_file", action="store"),
+    make_option(c("--segments_file", "-segments_file"), dest="segments_file", action="store"),
+    make_option(c("--output_dir", "-output_dir"), dest="output_dir", action="store"),
+    make_option(c("--sex_chrs", "-sexchrs"), dest="sex_chrs", action="store"))
+
+opt = parse_args(OptionParser(option_list=option_list))
+save(opt, file="debug.RData")
+
+sample_name=opt[["sample_name"]]
+snp_counts_file=opt[["snp_counts_file"]]
+coverage_file=opt[["coverage_file"]]
+segments_file=opt[["segments_file"]]
+output_file=opt[["output_dir"]]
+sex_chrs=as.logical(opt[["sex_chrs"]])
+num_chromosomes = ifelse(sex_chrs, 24, 22)
+
+#check input files exist.  If not, quit with error code that GATK will pick up
+if (!all(file.exists(c(snp_counts_file, coverage_file, segments_file)))) {
+    quit(save = "no", status = 1, runLast = FALSE)
+}
+
+create_acnv_plots_file = function(sample_name, snp_counts_file, coverage_file, segments_file, output_dir, num_chromosomes) {
+	#set up coverage, snps, and segments data frames
+	snp_counts = read.table(snp_counts_file, sep="\t", stringsAsFactors=FALSE, header=TRUE, check.names=FALSE)
+	segments = read.table(segments_file, sep="\t", stringsAsFactors=FALSE, header=TRUE, check.names=FALSE)
+    snp_counts[, "CONTIG"] = convert_XY_to_23_24(snp_counts[, "CONTIG"])
+    coverage = read.table(coverage_file, sep="\t", stringsAsFactors=FALSE, header=TRUE, check.names=FALSE)
+    names(coverage)[names(coverage) == sample_name] = "VALUE"
+    coverage$VALUE=2^coverage$VALUE #ACNV is in log space
+    segments[, "Chromosome"] = convert_XY_to_23_24(segments[, "Chromosome"])
+
+    #subset to autosomes if sex_chrs not specified
+    num_segs = length(which(segments[,"Chromosome"] <= num_chromosomes))
+    single_seg_chrs = c()
+    for(k in 1:num_chromosomes) {
+        if(sum(segments[,"Chromosome"] == k) == 1) {
+            single_seg_chrs = c(single_seg_chrs, k)
+        }
+    }
+    if(tail(single_seg_chrs, 1) == segments[nrow(segments), "Chromosome"]){ num_segs=num_segs+1 }
+    for( i in 1:(num_segs-1) ) {
+        seg1=segments[i,]
+        chr = seg1[, "Chromosome"]
+        only_1_seg = chr %in% single_seg_chrs
+        if(only_1_seg) {
```

Comment:
Instead of defining `seg2 = NULL` and `rbind`ing it,
```java
if (! chr %in% single_seg_chrs) {
    seg2 = segments[i + 1, ]
    if(chr != seg2[, "Chromosome"]) {next()}
    segs = rbind(seg1, seg2)
} else {
    segs = seg1
}
```